app/services/katello/pulp/repository/puppet.rb

```diff
@@ -40,8 +40,8 @@ def importer_class
 
         def copy_contents(destination_repo, options = {})
           if options[:puppet_modules]
-            module_uuids = options[:puppet_modules].pluck(:uuid)
-            clauses = { 'unit_id' => { "$in" => module_uuids } }
+            module_uuids = options[:puppet_modules].pluck(:pulp_id)
+            clauses = {'filters': { 'association': { 'unit_id' => { "$in" => module_uuids } } } }
```

Comment:
How did this work with the old clause?

Reply:
pulp ignores it, and it ends up copying everything in the source repository.  So this piece has been broken since https://projects.theforeman.org/issues/25233 was merged.    You only notice it if you actually look at the count of modules that end up in the view, or if the source repository has duplicate modules the distributor publish will fail (but neither of these happen with bats), i just got lucky on my testing :)

Reply:
:bat: